apps/mobile/src/context/NotificationsContext.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useContext, ReactNode } from 'react'
+
+import useNotifications from '@/src/hooks/useNotifications'
+import { FirebaseMessagingTypes } from '@react-native-firebase/messaging'
+
+interface NotificationContextType {
+  isDeviceNotificationEnabled: boolean
+  isAppNotificationEnabled: boolean
+  fcmToken: string | null
+  remoteMessages: FirebaseMessagingTypes.RemoteMessage[] | []
+}
+
+const NotificationContext = createContext<NotificationContextType | undefined>(undefined)
+
+export const useNotification = () => {
+  const context = useContext(NotificationContext)
+  if (context === undefined) {
+    throw new Error('useNotification must be used within a NotificationProvider')
+  }
+  return context
+}
+
+interface NotificationProviderProps {
+  children: ReactNode
+}
+
+export const NotificationsProvider: React.FC<NotificationProviderProps> = ({ children }) => {
```

Comment:
I guess a side effect of this provider is that we now immediately prompt the user for push notification permission when the app start. We should control the prompt somehow and first issue it later (on a specific screen).

Reply:
Now that I think of this. I guess we should not bother the user the first time he starts the app. Once he goes through the onboard there we have a screen to enable push notifications. If he denies there, then the next time he starts we can immeditely bother him again for the permissions (up to 3 times as you have it implemented it)